frontend/src/components/messages/NewRequestDialog.vue

```diff
@@ -547,12 +547,17 @@ export default {
     'newRequestModel.requestCategoryId': {
       async handler() {
         if (this.isIrregularExpenseRequest && !this.fundingAgreements) {
+          this.isLoading = true
           await this.getFundingAgreements()
+          this.isLoading = false
         }
         this.resetModelData(this.isAccountMaintenanceRequest)
         // Reset the Selected Facility if not allowed
         if (!isEmpty(this.newRequestModel.facilities) && !this.filteredFacilties?.some((fac) => fac.facilityId === this.newRequestModel.facilities[0]?.facilityId)) {
           this.newRequestModel.facilities = []
+          //then shorten the facility model to 1 item if going from multi-select to single select
+        } else if (this.newRequestModel.facilities?.length > 1 && (this.isAccountMaintenanceRequest || this.isIrregularExpenseRequest || this.isReportingRequest)) {
```

Comment:
can we use !isMultipleFacilities() in this logic?

Reply:
yup!! great call